ibc/packet.go

```diff
@@ -0,0 +1,58 @@
+package ibc
+
+import (
+	"errors"
+	"fmt"
+
+	host "github.com/cosmos/ibc-go/v3/modules/core/24-host"
+	"go.uber.org/multierr"
+)
+
+type Nanoseconds uint64
+
+// Packet is a packet sent over an IBC channel as defined in ICS-4.
+// See: https://github.com/cosmos/ibc/blob/52a9094a5bc8c5275e25c19d0b2d9e6fd80ba31c/spec/core/ics-004-channel-and-packet-semantics/README.md
+// Proto defined at: github.com/cosmos/ibc-go/v3@v3.0.0/proto/ibc/core/channel/v1/tx.proto
+type Packet struct {
+	Sequence      uint64 // the order of sends and receives, where a packet with an earlier sequence number must be sent and received before a packet with a later sequence number
+	SourcePort    string // the port on the sending chain
+	SourceChannel string // the channel end on the sending chain
+	DestPort      string // the port on the receiving chain
+	DestChannel   string // the channel end on the receiving chain
+	Data          []byte // an opaque value which can be defined by the application logic of the associated modules
+	TimeoutHeight string // a consensus height on the destination chain after which the packet will no longer be processed, and will instead count as having timed-out
+
+	// Indicates a timestamp (in nanoseconds) on the destination chain after which the packet will no longer be processed, and will instead count as having timed-out.
+	// The IBC spec does not indicate the unit of time. However, ibc-go's protos define it as nanoseconds.
+	TimeoutTimestamp Nanoseconds
+}
+
+// Validate returns an error if the packet is not well-formed.
+func (packet Packet) Validate() error {
+	var merr error
+	addErr := func(err error) {
+		merr = multierr.Append(merr, err)
+	}
```

Comment:
This local function seems unnecessary. But it will probably get inlined during compilation.

```go
// Either of these should be equivalent.
merr = multierr.Append(merr, fmt.Errorf("..."))
multierr.AppendInto(&merr, fmt.Errorf("..."))
```

Reply:
Stylistically, I felt the local function was easier to read and write, although I was not aware `AppendInto` was part of the package api.